ros/processor/suggestions_engine.py

```diff
@@ -1,26 +1,93 @@
-import time
+import json
+import requests
+from http import HTTPStatus
+from ros.lib import consume
 from ros.lib.config import get_logger
+from tempfile import NamedTemporaryFile
 from prometheus_client import start_http_server
-from ros.lib.config import ROS_SUGGESTIONS_ENGINE_PORT
+from ros.lib.config import INVENTORY_EVENTS_TOPIC, METRICS_PORT
 
 
 logging = get_logger(__name__)
 
 
 class SuggestionsEngine:
     def __init__(self):
+        self.consumer = consume.init_consumer(INVENTORY_EVENTS_TOPIC)
+        self.service = 'SUGGESTIONS_ENGINE'
+        self.event = None
+
+    def handle_create_update(self, payload):
+        self.event = "Update event" if payload.get('type') == 'updated' else "Create event"
+
+        platform_metadata = payload.get('platform_metadata')
+        host = payload.get('host')
+
+        if platform_metadata is None or host is None:
+            logging.info(f"{self.service} - {self.event} - Missing host or/and platform_metadata field(s).")
+            return
+
+        if not is_pcp_collected(platform_metadata):
+            return
+
+        archiveURL = platform_metadata.get('url')
+
+        logging.info(f"{self.service} - {self.event} - Downloading the report for system {host.get('id')}.")
+
+        response = requests.get(archiveURL, timeout=10)
+
+        if response.status_code != HTTPStatus.OK:
+            logging.error(
+                f"{self.service} - {self.event} - Unable to download the report for system {host.get('id')}. "
+                f"ERROR - {response.reason}"
+            )
+        else:
+            with NamedTemporaryFile(delete=True) as tempfile:
+                tempfile.write(response.content)
```

Comment:
This is just downloading file and writing it to temp file, are we going to have separate pull request for extracting the archive?

Is it possible to separate out logic about downloading, extracting the archive?